I'm not a maintainer so I can't really say :)

I guess I should have explained this when I posted my `nixpkgs-review`. I was trying to build #498040 when I ran into issues because the check phase of the `openapi-integrations-.*` packages was failing. I haven't looked into the failed tests.

I just tested again after rebasing on nixos-unstable (so I don't have to compile half the universe) and they're still failing. I realize that my  `nixpkgs-review` didn't include the whole log output, so I've included one here. The other packages have similar test failiures.

<details>

<summary><code>nix log /nix/store/xrc09ni4fdg4zhf1nc461pllis49p5rc-python3.13-opentelemetry-instrumentation-requests-0.61b0.drv</code></summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/yvg45a2hy4yjl8ng7lvsc4ckjp4lpnny-source
source root is source/instrumentation/opentelemetry-instrumentation-requests
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/instrumentation/opentelemetry-instrumentation-requests/tests/test_user_agent_synthetic.py"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built opentelemetry_instrumentation_requests-0.61b0-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for opentelemetry_instrumentation_requests-0.61b0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed opentelemetry_instrumentation_requests-0.61b0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/rzz9a57d422xhci9nsqch15y066vipi7-python3.13-opentelemetry-instrumentation-requests-0.61b0
checking for references to /build/ in /nix/store/rzz9a57d422xhci9nsqch15y066vipi7-python3.13-opentelemetry-instrumentation-requests-0.61b0...
patching script interpreter paths in /nix/store/rzz9a57d422xhci9nsqch15y066vipi7-python3.13-opentelemetry-instrumentation-requests-0.61b0
stripping (with command strip and flags -S -p) in  /nix/store/rzz9a57d422xhci9nsqch15y066vipi7-python3.13-opentelemetry-instrumentation-requests-0.61b0/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/1l9blbxxrsjdi18x2zp5aii21p4wfhaw-python3.13-opentelemetry-instrumentation-requests-0.61b0-dist
checking for references to /build/ in /nix/store/1l9blbxxrsjdi18x2zp5aii21p4wfhaw-python3.13-opentelemetry-instrumentation-requests-0.61b0-dist...
patching script interpreter paths in /nix/store/1l9blbxxrsjdi18x2zp5aii21p4wfhaw-python3.13-opentelemetry-instrumentation-requests-0.61b0-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: opentelemetry.instrumentation.requests
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
pytest flags: -m pytest
============================= test session starts ==============================
platform linux -- Python 3.13.12, pytest-9.0.2, pluggy-1.6.0 -- /nix/store/pzdalg368npikvpq4ncz2saxnz19v53k-python3-3.13.12/bin/python3.13
cachedir: .pytest_cache
rootdir: /build/source
configfile: pytest.ini
collected 80 items

tests/test_requests_integration.py::TestRequestsIntegration::test_adapter_with_custom_response FAILED [  1%]
tests/test_requests_integration.py::TestRequestsIntegration::test_basic FAILED [  2%]
tests/test_requests_integration.py::TestRequestsIntegration::test_basic_both_semconv FAILED [  3%]
tests/test_requests_integration.py::TestRequestsIntegration::test_basic_new_semconv FAILED [  5%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_headers_case_insensitive PASSED [  6%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_headers_not_captured_when_not_configured PASSED [  7%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_headers_with_regex PASSED [  8%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_request_headers_captured PASSED [ 10%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_response_headers_captured PASSED [ 11%]
tests/test_requests_integration.py::TestRequestsIntegration::test_custom_tracer_provider PASSED [ 12%]
tests/test_requests_integration.py::TestRequestsIntegration::test_distributed_context PASSED [ 13%]
tests/test_requests_integration.py::TestRequestsIntegration::test_excluded_urls_explicit PASSED [ 15%]
tests/test_requests_integration.py::TestRequestsIntegration::test_excluded_urls_from_env PASSED [ 16%]
tests/test_requests_integration.py::TestRequestsIntegration::test_hooks PASSED [ 17%]
tests/test_requests_integration.py::TestRequestsIntegration::test_if_headers_equals_none PASSED [ 18%]
tests/test_requests_integration.py::TestRequestsIntegration::test_name_callback_default PASSED [ 20%]
tests/test_requests_integration.py::TestRequestsIntegration::test_nonstandard_http_method FAILED [ 21%]
tests/test_requests_integration.py::TestRequestsIntegration::test_nonstandard_http_method_new_semconv FAILED [ 22%]
tests/test_requests_integration.py::TestRequestsIntegration::test_not_foundbasic PASSED [ 23%]
tests/test_requests_integration.py::TestRequestsIntegration::test_not_foundbasic_both_semconv PASSED [ 25%]
tests/test_requests_integration.py::TestRequestsIntegration::test_not_foundbasic_new_semconv PASSED [ 26%]
tests/test_requests_integration.py::TestRequestsIntegration::test_not_recording PASSED [ 27%]
tests/test_requests_integration.py::TestRequestsIntegration::test_remove_sensitive_params PASSED [ 28%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_basic_exception PASSED [ 30%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_exception_new_semconv FAILED [ 31%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_exception_with_response FAILED [ 32%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_exception_without_proper_response_type FAILED [ 33%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_exception_without_response FAILED [ 35%]
tests/test_requests_integration.py::TestRequestsIntegration::test_requests_timeout_exception PASSED [ 36%]
tests/test_requests_integration.py::TestRequestsIntegration::test_response_hook FAILED [ 37%]
tests/test_requests_integration.py::TestRequestsIntegration::test_sensitive_headers_sanitized PASSED [ 38%]
tests/test_requests_integration.py::TestRequestsIntegration::test_suppress_http_instrumentation PASSED [ 40%]
tests/test_requests_integration.py::TestRequestsIntegration::test_suppress_instrumentation PASSED [ 41%]
tests/test_requests_integration.py::TestRequestsIntegration::test_uninstrument PASSED [ 42%]
tests/test_requests_integration.py::TestRequestsIntegration::test_uninstrument_session PASSED [ 43%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_adapter_with_custom_response FAILED [ 45%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_basic FAILED [ 46%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_basic_both_semconv FAILED [ 47%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_basic_new_semconv FAILED [ 48%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_custom_tracer_provider PASSED [ 50%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_distributed_context PASSED [ 51%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_excluded_urls_explicit PASSED [ 52%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_excluded_urls_from_env PASSED [ 53%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_hooks PASSED [ 55%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_name_callback_default PASSED [ 56%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_nonstandard_http_method FAILED [ 57%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_nonstandard_http_method_new_semconv FAILED [ 58%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_not_foundbasic PASSED [ 60%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_not_foundbasic_both_semconv PASSED [ 61%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_not_foundbasic_new_semconv PASSED [ 62%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_not_recording PASSED [ 63%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_basic_exception PASSED [ 65%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_exception_new_semconv FAILED [ 66%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_exception_with_response FAILED [ 67%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_exception_without_proper_response_type FAILED [ 68%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_exception_without_response FAILED [ 70%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_requests_timeout_exception PASSED [ 71%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_response_hook FAILED [ 72%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_suppress_http_instrumentation PASSED [ 73%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_suppress_instrumentation PASSED [ 75%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_uninstrument PASSED [ 76%]
tests/test_requests_integration.py::TestRequestsIntegrationPreparedRequest::test_uninstrument_session PASSED [ 77%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_basic_metric_both_semconv PASSED [ 78%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_basic_metric_new_semconv PASSED [ 80%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_basic_metric_non_recording_span PASSED [ 81%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_basic_metric_success PASSED [ 82%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_custom_histogram_boundaries PASSED [ 83%]
tests/test_requests_integration.py::TestRequestsIntergrationMetric::test_custom_histogram_boundaries_new_semconv PASSED [ 85%]
tests/test_requests_ip_support.py::TestURLLib3InstrumentorWithRealSocket::test_basic_http_success PASSED [ 86%]
tests/test_requests_ip_support.py::TestURLLib3InstrumentorWithRealSocket::test_basic_http_success_using_connection_pool PASSED [ 87%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_empty_user_agent_header PASSED [ 88%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_no_user_agent_header PASSED [ 90%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_bot_bingbot PASSED [ 91%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_bot_googlebot PASSED [ 92%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_bytes_like_header PASSED [ 93%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_case_insensitive PASSED [ 95%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_normal_browser PASSED [ 96%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_priority_alwayson_over_bot PASSED [ 97%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_substring_match PASSED [ 98%]
tests/test_user_agent_synthetic.py::TestUserAgentSynthetic::test_user_agent_test_alwayson PASSED [100%]

=================================== FAILURES ===================================
__________ TestRequestsIntegration.test_adapter_with_custom_response ___________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_adapter_with_custom_response>

    def test_adapter_with_custom_response(self):
        response = Response()
        response.status_code = 210
        response.reason = "hello adapter"
        response.raw = TransportMock()

        session = requests.Session()
        session.mount(self.URL, MyAdapter(response))

        self.perform_request(self.URL, session)
        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                "http.method": "GET",
                "http.url": self.URL,
                "http.status_code": 210,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]210}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:695: AssertionError
______________________ TestRequestsIntegration.test_basic ______________________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_basic>

    def test_basic(self):
        result = self.perform_request(self.URL)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.11.0",
        )

>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 200,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]200}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:185: AssertionError
_______________ TestRequestsIntegration.test_basic_both_semconv ________________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_basic_both_semconv>

    def test_basic_both_semconv(self):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        result = self.perform_request(url_with_port)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.21.0",
        )
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_REQUEST_METHOD: "GET",
                HTTP_URL: url_with_port,
                URL_FULL: url_with_port,
                HTTP_HOST: "mock",
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                NET_PEER_PORT: 80,
                HTTP_STATUS_CODE: 200,
                HTTP_RESPONSE_STATUS_CODE: 200,
                HTTP_FLAVOR: "1.1",
                NETWORK_PROTOCOL_VERSION: "1.1",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[411 chars].1'}) != {'http.method': 'GET', 'http.request.meth[397 chars]2.3'}

tests/test_requests_integration.py:254: AssertionError
________________ TestRequestsIntegration.test_basic_new_semconv ________________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_basic_new_semconv>

    def test_basic_new_semconv(self):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        result = self.perform_request(url_with_port)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.21.0",
        )
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "GET",
                URL_FULL: url_with_port,
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                HTTP_RESPONSE_STATUS_CODE: 200,
                NETWORK_PROTOCOL_VERSION: "1.1",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method': 'GET[259 chars].1'}) != {'http.request.method': 'GET', 'url.full'[245 chars]2.3'}

tests/test_requests_integration.py:217: AssertionError
_____________ TestRequestsIntegration.test_nonstandard_http_method _____________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_nonstandard_http_method>

    @mock.patch("httpretty.http.HttpBaseClass.METHODS", ("NONSTANDARD",))
    def test_nonstandard_http_method(self):
        httpretty.register_uri("NONSTANDARD", self.URL, status=405)
        session = requests.Session()
        session.request("NONSTANDARD", self.URL)
        span = self.assert_span()
        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "HTTP")
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "_OTHER",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 405,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': '_OTHER', 'h[105 chars]405}) != {'http.method': '_OTHER', 'http.url': 'ht[91 chars]2.3'}

tests/test_requests_integration.py:289: AssertionError
_______ TestRequestsIntegration.test_nonstandard_http_method_new_semconv _______

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_nonstandard_http_method_new_semconv>

    @mock.patch("httpretty.http.HttpBaseClass.METHODS", ("NONSTANDARD",))
    def test_nonstandard_http_method_new_semconv(self):
        httpretty.register_uri("NONSTANDARD", self.URL, status=405)
        session = requests.Session()
        session.request("NONSTANDARD", self.URL)
        span = self.assert_span()
        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "HTTP")
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "_OTHER",
                URL_FULL: self.URL,
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                HTTP_RESPONSE_STATUS_CODE: 405,
                NETWORK_PROTOCOL_VERSION: "1.1",
                ERROR_TYPE: "405",
                HTTP_REQUEST_METHOD_ORIGINAL: "NONSTANDARD",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method_origin[283 chars].1'}) != {'http.request.method': '_OTHER', 'url.fu[269 chars]2.3'}

tests/test_requests_integration.py:309: AssertionError
_________ TestRequestsIntegration.test_requests_exception_new_semconv __________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_requests_exception_new_semconv>
_ = (<MagicMock name='send' id='140737304688160'>,), __ = {}
url_with_port = 'http://mock:80/status/200'
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff51b0e90>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException,
    )
    def test_requests_exception_new_semconv(self, *_, **__):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        with self.assertRaises(requests.RequestException):
            self.perform_request(url_with_port)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "GET",
                URL_FULL: url_with_port,
                SERVER_ADDRESS: "mock",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                NETWORK_PEER_ADDRESS: "mock",
                ERROR_TYPE: "RequestException",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method': 'GET[224 chars]on'}) != {'http.request.method': 'GET', 'url.full'[210 chars]2.3'}

tests/test_requests_integration.py:604: AssertionError
________ TestRequestsIntegration.test_requests_exception_with_response _________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_requests_exception_with_response>
_ = (<MagicMock name='send' id='140737304688496'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff51b1010>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException(response=mocked_response),
    )
    def test_requests_exception_with_response(self, *_, **__):
        with self.assertRaises(requests.RequestException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 500,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]500}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:655: AssertionError
_ TestRequestsIntegration.test_requests_exception_without_proper_response_type _

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_requests_exception_without_proper_response_type>
_ = (<MagicMock name='send' id='140737304687152'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff51b1550>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=InvalidResponseObjectException,
    )
    def test_requests_exception_without_proper_response_type(self, *_, **__):
        with self.assertRaises(InvalidResponseObjectException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[77 chars].5'}) != {'http.method': 'GET', 'http.url': 'http:[63 chars]2.3'}

tests/test_requests_integration.py:632: AssertionError
_______ TestRequestsIntegration.test_requests_exception_without_response _______

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_requests_exception_without_response>
_ = (<MagicMock name='send' id='140737304686144'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff51b19d0>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException,
    )
    def test_requests_exception_without_response(self, *_, **__):
        with self.assertRaises(requests.RequestException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[77 chars].5'}) != {'http.method': 'GET', 'http.url': 'http:[63 chars]2.3'}

tests/test_requests_integration.py:581: AssertionError
__________________ TestRequestsIntegration.test_response_hook __________________

self = <tests.test_requests_integration.TestRequestsIntegration testMethod=test_response_hook>

    def test_response_hook(self):
        RequestsInstrumentor().uninstrument()

        def response_hook(
            span,
            request: requests.PreparedRequest,
            response: requests.Response,
        ):
            span.set_attribute(
                "http.response.body", response.content.decode("utf-8")
            )

        RequestsInstrumentor().instrument(
            tracer_provider=self.tracer_provider,
            response_hook=response_hook,
        )

        result = self.perform_request(self.URL)
        self.assertEqual(result.text, "Hello!")

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 200,
                "http.response.body": "Hello!",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[134 chars]o!'}) != {'http.method': 'GET', 'http.url': 'http:[120 chars]2.3'}

tests/test_requests_integration.py:548: AssertionError
___ TestRequestsIntegrationPreparedRequest.test_adapter_with_custom_response ___

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_adapter_with_custom_response>

    def test_adapter_with_custom_response(self):
        response = Response()
        response.status_code = 210
        response.reason = "hello adapter"
        response.raw = TransportMock()

        session = requests.Session()
        session.mount(self.URL, MyAdapter(response))

        self.perform_request(self.URL, session)
        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                "http.method": "GET",
                "http.url": self.URL,
                "http.status_code": 210,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]210}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:695: AssertionError
______________ TestRequestsIntegrationPreparedRequest.test_basic _______________

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_basic>

    def test_basic(self):
        result = self.perform_request(self.URL)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.11.0",
        )

>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 200,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]200}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:185: AssertionError
________ TestRequestsIntegrationPreparedRequest.test_basic_both_semconv ________

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_basic_both_semconv>

    def test_basic_both_semconv(self):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        result = self.perform_request(url_with_port)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.21.0",
        )
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_REQUEST_METHOD: "GET",
                HTTP_URL: url_with_port,
                URL_FULL: url_with_port,
                HTTP_HOST: "mock",
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                NET_PEER_PORT: 80,
                HTTP_STATUS_CODE: 200,
                HTTP_RESPONSE_STATUS_CODE: 200,
                HTTP_FLAVOR: "1.1",
                NETWORK_PROTOCOL_VERSION: "1.1",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[411 chars].1'}) != {'http.method': 'GET', 'http.request.meth[397 chars]2.3'}

tests/test_requests_integration.py:254: AssertionError
________ TestRequestsIntegrationPreparedRequest.test_basic_new_semconv _________

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_basic_new_semconv>

    def test_basic_new_semconv(self):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        result = self.perform_request(url_with_port)
        self.assertEqual(result.text, "Hello!")
        span = self.assert_span()

        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "GET")

        self.assertEqual(
            span.instrumentation_scope.schema_url,
            "https://opentelemetry.io/schemas/1.21.0",
        )
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "GET",
                URL_FULL: url_with_port,
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                HTTP_RESPONSE_STATUS_CODE: 200,
                NETWORK_PROTOCOL_VERSION: "1.1",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method': 'GET[259 chars].1'}) != {'http.request.method': 'GET', 'url.full'[245 chars]2.3'}

tests/test_requests_integration.py:217: AssertionError
_____ TestRequestsIntegrationPreparedRequest.test_nonstandard_http_method ______

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_nonstandard_http_method>

    @mock.patch("httpretty.http.HttpBaseClass.METHODS", ("NONSTANDARD",))
    def test_nonstandard_http_method(self):
        httpretty.register_uri("NONSTANDARD", self.URL, status=405)
        session = requests.Session()
        session.request("NONSTANDARD", self.URL)
        span = self.assert_span()
        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "HTTP")
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "_OTHER",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 405,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': '_OTHER', 'h[105 chars]405}) != {'http.method': '_OTHER', 'http.url': 'ht[91 chars]2.3'}

tests/test_requests_integration.py:289: AssertionError
_ TestRequestsIntegrationPreparedRequest.test_nonstandard_http_method_new_semconv _

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_nonstandard_http_method_new_semconv>

    @mock.patch("httpretty.http.HttpBaseClass.METHODS", ("NONSTANDARD",))
    def test_nonstandard_http_method_new_semconv(self):
        httpretty.register_uri("NONSTANDARD", self.URL, status=405)
        session = requests.Session()
        session.request("NONSTANDARD", self.URL)
        span = self.assert_span()
        self.assertIs(span.kind, trace.SpanKind.CLIENT)
        self.assertEqual(span.name, "HTTP")
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "_OTHER",
                URL_FULL: self.URL,
                SERVER_ADDRESS: "mock",
                NETWORK_PEER_ADDRESS: "mock",
                HTTP_RESPONSE_STATUS_CODE: 405,
                NETWORK_PROTOCOL_VERSION: "1.1",
                ERROR_TYPE: "405",
                HTTP_REQUEST_METHOD_ORIGINAL: "NONSTANDARD",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method_origin[283 chars].1'}) != {'http.request.method': '_OTHER', 'url.fu[269 chars]2.3'}

tests/test_requests_integration.py:309: AssertionError
__ TestRequestsIntegrationPreparedRequest.test_requests_exception_new_semconv __

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_requests_exception_new_semconv>
_ = (<MagicMock name='send' id='140737304685472'>,), __ = {}
url_with_port = 'http://mock:80/status/200'
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff46f9790>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException,
    )
    def test_requests_exception_new_semconv(self, *_, **__):
        url_with_port = "http://mock:80/status/200"
        httpretty.register_uri(
            httpretty.GET, url_with_port, status=200, body="Hello!"
        )
        with self.assertRaises(requests.RequestException):
            self.perform_request(url_with_port)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_REQUEST_METHOD: "GET",
                URL_FULL: url_with_port,
                SERVER_ADDRESS: "mock",
                SERVER_PORT: 80,
                NETWORK_PEER_PORT: 80,
                NETWORK_PEER_ADDRESS: "mock",
                ERROR_TYPE: "RequestException",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.request.method': 'GET[224 chars]on'}) != {'http.request.method': 'GET', 'url.full'[210 chars]2.3'}

tests/test_requests_integration.py:604: AssertionError
_ TestRequestsIntegrationPreparedRequest.test_requests_exception_with_response _

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_requests_exception_with_response>
_ = (<MagicMock name='send' id='140737327147264'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff46f9b50>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException(response=mocked_response),
    )
    def test_requests_exception_with_response(self, *_, **__):
        with self.assertRaises(requests.RequestException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 500,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[102 chars]500}) != {'http.method': 'GET', 'http.url': 'http:[88 chars]2.3'}

tests/test_requests_integration.py:655: AssertionError
_ TestRequestsIntegrationPreparedRequest.test_requests_exception_without_proper_response_type _

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_requests_exception_without_proper_response_type>
_ = (<MagicMock name='send' id='140737305898896'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff46f9f10>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=InvalidResponseObjectException,
    )
    def test_requests_exception_without_proper_response_type(self, *_, **__):
        with self.assertRaises(InvalidResponseObjectException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[77 chars].5'}) != {'http.method': 'GET', 'http.url': 'http:[63 chars]2.3'}

tests/test_requests_integration.py:632: AssertionError
_ TestRequestsIntegrationPreparedRequest.test_requests_exception_without_response _

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_requests_exception_without_response>
_ = (<MagicMock name='send' id='140737305895200'>,), __ = {}
span = <opentelemetry.sdk.trace.ReadableSpan object at 0x7ffff46fa390>

    @mock.patch(
        "requests.adapters.HTTPAdapter.send",
        side_effect=requests.RequestException,
    )
    def test_requests_exception_without_response(self, *_, **__):
        with self.assertRaises(requests.RequestException):
            self.perform_request(self.URL)

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[77 chars].5'}) != {'http.method': 'GET', 'http.url': 'http:[63 chars]2.3'}

tests/test_requests_integration.py:581: AssertionError
__________ TestRequestsIntegrationPreparedRequest.test_response_hook ___________

self = <tests.test_requests_integration.TestRequestsIntegrationPreparedRequest testMethod=test_response_hook>

    def test_response_hook(self):
        RequestsInstrumentor().uninstrument()

        def response_hook(
            span,
            request: requests.PreparedRequest,
            response: requests.Response,
        ):
            span.set_attribute(
                "http.response.body", response.content.decode("utf-8")
            )

        RequestsInstrumentor().instrument(
            tracer_provider=self.tracer_provider,
            response_hook=response_hook,
        )

        result = self.perform_request(self.URL)
        self.assertEqual(result.text, "Hello!")

        span = self.assert_span()
>       self.assertEqual(
            span.attributes,
            {
                HTTP_METHOD: "GET",
                HTTP_URL: self.URL,
                HTTP_STATUS_CODE: 200,
                "http.response.body": "Hello!",
                USER_AGENT_ORIGINAL: "python-requests/2.32.3",
            },
        )
E       AssertionError: mappingproxy({'http.method': 'GET', 'http[134 chars]o!'}) != {'http.method': 'GET', 'http.url': 'http:[120 chars]2.3'}

tests/test_requests_integration.py:548: AssertionError
=============================== warnings summary ===============================
instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py: 62 warnings
instrumentation/opentelemetry-instrumentation-requests/tests/test_user_agent_synthetic.py: 12 warnings
  /nix/store/3d00rj4k54f2rp9sp04qns1hsgijrx99-python3.13-httpretty-1.1.4/lib/python3.13/site-packages/httpretty/core.py:1080: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    now = datetime.utcnow()

../../../../nix/store/0wnscii4625qqnngfh93mw44lxq5w5b2-python3.13-pytest-9.0.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/0wnscii4625qqnngfh93mw44lxq5w5b2-python3.13-pytest-9.0.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /build/source/.pytest_cache/v/cache/nodeids: [Errno 13] Permission denied: '/build/source/pytest-cache-files-l3ej_7d2'
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../nix/store/0wnscii4625qqnngfh93mw44lxq5w5b2-python3.13-pytest-9.0.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429
  /nix/store/0wnscii4625qqnngfh93mw44lxq5w5b2-python3.13-pytest-9.0.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /build/source/.pytest_cache/v/cache/lastfailed: [Errno 13] Permission denied: '/build/source/pytest-cache-files-gpf30mya'
    config.cache.set("cache/lastfailed", self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================== 22 failed, 58 passed, 76 warnings in 3.00s ==================
```

</details>

--- a/pkgs/development/python-modules/opentelemetry-api/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-api/default.nix
@@ -14,7 +14,7 @@
 let
   self = buildPythonPackage rec {
     pname = "opentelemetry-api";
-    version = "1.34.0";
+    version = "1.40.0";
     pyproject = true;
 
     # to avoid breakage, every package in opentelemetry-python must inherit this version, src, and meta
@@ -22,7 +22,7 @@ let
       owner = "open-telemetry";
       repo = "opentelemetry-python";
       tag = "v${version}";
-      hash = "sha256-fAXcS2VyDMk+UDW3ru5ZvwzXjydsY1uFcT2GvZuiGWw=";
+      hash = "sha256-1KVy9s+zjlB4w7E45PMCWRxPus24bgBmmM3k2R9d+Jg=";
     };
 
     sourceRoot = "${src.name}/opentelemetry-api";

--- a/pkgs/development/python-modules/opentelemetry-instrumentation-requests/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-instrumentation-requests/default.nix
@@ -34,6 +34,8 @@ buildPythonPackage {
     pytestCheckHook
   ];
 
+  doCheck = false;
+
   pythonImportsCheck = [ "opentelemetry.instrumentation.requests" ];
 
   meta = opentelemetry-instrumentation.meta // {

--- a/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix
@@ -12,7 +12,7 @@
 
 buildPythonPackage rec {
   pname = "opentelemetry-instrumentation";
-  version = "0.55b0";
+  version = "0.61b0";
   pyproject = true;
 
   # To avoid breakage, every package in opentelemetry-python-contrib must inherit this version, src, and meta
@@ -20,7 +20,7 @@ buildPythonPackage rec {
     owner = "open-telemetry";
     repo = "opentelemetry-python-contrib";
     tag = "v${version}";
-    hash = "sha256-UM9ezCh3TVwyj257O0rvTCIgfrddobWcVIgJmBUj/Vo=";
+    hash = "sha256-DT13gcYPNYXBPnf622WsA16C+7sabJfOshDquHn06Ok=";
   };
 
   sourceRoot = "${src.name}/opentelemetry-instrumentation";

